ogr/ogrsf_frmts/generic/ogrlayer.cpp

```diff
@@ -1439,13 +1439,55 @@ OGRGeometryH OGR_L_GetSpatialFilter(OGRLayerH hLayer)
         OGRLayer::FromHandle(hLayer)->GetSpatialFilter());
 }
 
+/************************************************************************/
+/*             ValidateGeometryFieldIndexForSetSpatialFilter()          */
+/************************************************************************/
+
+//! @cond Doxygen_Suppress
+bool OGRLayer::ValidateGeometryFieldIndexForSetSpatialFilter(
+    int iGeomField, const OGRGeometry *poGeomIn, bool bIsSelectLayer)
+{
+    if (iGeomField == 0 && poGeomIn == nullptr &&
+        GetLayerDefn()->GetGeomFieldCount() == 0)
+    {
+        /* do nothing */
```

Comment:
shouldn't this return false?

Reply:
no, this is a tolerance. Setting a null spatial filter on geometry field idx 0 when there are no geometry field can't harm, and is accepted silently for backward compatibility with existing practice. I'm adding a code comment to make that clearer